src/Microsoft.AspNet.Mvc.Localization/LocalizedHtmlString.cs

```diff
@@ -41,11 +57,145 @@ public LocalizedHtmlString(string key, string value, bool isResourceNotFound)
+            if (htmlTextWriter == null)
+            {
+                FormatValue(writer, encoder, Value, _arguments);
+            }
+            else
+            {
+                htmlTextWriter.Write(this);
+            }
+        }
+
+        private void FormatValue(
+            TextWriter writer,
+            HtmlEncoder encoder,
+            string resourceString,
+            object[] arguments)
+        {
+            var position = 0;
+            var length = resourceString.Length;
+            StringBuilder tokenBuffer = null;
+            var isToken = false;
+
+            while (position < length)
+            {
+                var currentCharacter = resourceString[position];
+                position++;
+
+                if (currentCharacter == '}')
+                {
+                    if (position < length && resourceString[position] == '}')
+                    {
+                        // Escaped curly brace: "}}".
+                        AppendCurly(isToken, currentCharacter, tokenBuffer, writer);
+                        position++;
+                    }
+                    else
+                    {
+                        // End of a token.
+                        Append(isToken, '}', tokenBuffer, writer);
+                        if (position == length)
+                        {
+                            break;
+                        }
+
+                        AppendToOutput(tokenBuffer, arguments, writer, encoder);
+
+                        isToken = false;
+                        tokenBuffer = null;
+                    }
+                }
+                else if (currentCharacter == '{')
+                {
+                    if (position < length && resourceString[position] == '{')
+                    {
+                        // Escaped curly brace: "{{".
+                        AppendCurly(isToken, currentCharacter, tokenBuffer, writer);
+                        position++;
+                    }
+                    else
+                    {
+                        // Start of a new token.
+                        tokenBuffer = new StringBuilder();
+                        tokenBuffer.Append('{');
+                        isToken = true;
+                    }
+                }
+                else
+                {
+                    Append(isToken, currentCharacter, tokenBuffer, writer);
+                }
+            }
+
+            AppendToOutput(tokenBuffer, arguments, writer, encoder);
+        }
+
+        private void Append(
+            bool isToken,
+            char value,
+            StringBuilder tokenBuffer,
+            TextWriter writer)
+        {
+            if (isToken)
+            {
+                tokenBuffer.Append(value);
+            }
+            else
+            {
+                writer.Write(value);
+            }
+        }
+
+        private void AppendCurly(
+            bool isToken,
+            char curly,
```

Comment:
Not super vested. `curly` reminds me of a comedy act rather than the serious business that is localization.